### PR TITLE
USWDS - Footer: Fix overflow focus cutoff and padding

### LINKED ISSUE #6092: USWDS - Bug: Focus ring in footer links is cut off
### Describe the bug

The focus ring around footer links is cut off.

Image

### Steps to reproduce the bug

1. Create a footer using code like so:
```html
<!DOCTYPE html>
<html>
<head>
  <link href="https://cdn.jsdelivr.net/npm/@uswds/uswds@3.9.0/dist/css/uswds.min.css" rel="stylesheet">
</head>
<body class="display-flex flex-column" style="min-height: 100vh;">
  <main class="flex-1"></main>
  <footer class="usa-footer usa-footer--medium">
    <div class="usa-footer__primary-section">
      <nav class="usa-footer__nav">
        <ul class="grid-row grid-gap">
          <li class="desktop:grid-col-auto usa-footer__primary-content mobile-lg:grid-col-4">
            <a class="usa-footer__primary-link" href="#">Test</a>
          </li>
          <li class="desktop:grid-col-auto usa-footer__primary-content mobile-lg:grid-col-4">
            <a class="usa-footer__primary-link" href="#">Test</a>
          </li>
        </ul>
      </nav>
    </div>
  </footer>
</body>
</html>
```
2. Observe that the top half of the focus ring is cut off

### Expected Behavior

The focus ring should show fully around the links in the footer.

Image

### Related code

_No response_

### Screenshots

Current behavior:

Image

Expected behavior:

Image

### System setup

- USWDS version: 3.9.0
- browser: all major browsers

### Additional context

_No response_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary
**Removed `overflow: hidden` from `usa-footer` to allow the full focus outline to show.** This fix also improves alignment in the slim footer variant. 

> [!note]
> This is a follow-on for PR #6093 (Thanks @6TELOIV!)

## Breaking change

This is not a breaking change.

## Related issue

Closes #6092

## Related pull requests

[Changelog](https://github.com/uswds/uswds-site/pull/2999)

## Preview link

[Footer component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-footer-overflow-focus/?path=/story/components-footer--slim-footer)

## Problem statement
Focus rings are visually cut off in the footer. Additionally, the padding in the slim footer prevents the top border from extending the full width of the component (like it does in the other variants). 
image

## Solution
1. Removing `overflow: hidden` allows focus rings to show at the edge of the footer.
1. Removing the x padding from the slim footer mobile presentation resolves both of these issues. 

#### Default footer

image

#### Slim footer
image

Note that this change also fixes the slim footer alignment in medium screen widths: 

image

## Testing and review
- Footer no longer hides pieces of the focus outline
- Confirm there are no visual regressions:
   - As a result of removing `overflow: hidden`
   - In link lists in all variants in mobile view
   - In link lists in all variants in desktop view
- Confirm code meets USWDS convention

### CODE REVIEW

File: packages/usa-footer/src/styles/_usa-footer.scss [hunk 1]
```diff
@@ -66,6 +65,7 @@ $-chevron-expand-more: map-merge(
 
 .usa-footer__primary-container {
   @include grid-container($theme-footer-max-width);
+  @include u-padding-x(0);
```
Comment (amyleadem) [inline packages/usa-footer/src/styles/_usa-footer.scss:68]: > [!note]
> This is the only meaningful change from #6093. All other changes are just re-ordering the style rules to match convention.

**Review by mejiaj — APPROVED** [review]
Confirming that all variants look good on all screen sizes. Thanks for building off of the good work from @6TELOIV.

**Review by mahoneycm — APPROVED** [review]
This looks great! All three variants are much more uniform now. 

**Question:** Should we set the target to #6093? Or are we going to close that one in favor of this one? 

It might be good to keep the previous one due to the discussion and decisions captured there

File: packages/usa-footer/src/styles/_usa-footer.scss (hunk 1 above)
Comment (6TELOIV) [reply packages/usa-footer/src/styles/_usa-footer.scss:68]: I think this looks good to me! Both issues are fairly related, so getting both done in one pull makes sense to me

**Review by thisisdano — APPROVED** [review]

**Pull request merged**